server/sonar-web/src/main/js/apps/projectBranches/components/App.tsx

```diff
@@ -19,42 +19,129 @@
  */
 import * as React from 'react';
 import BranchRow from './BranchRow';
+import LongBranchesPattern from './LongBranchesPattern';
 import { Branch } from '../../../app/types';
 import { sortBranchesAsTree } from '../../../helpers/branches';
 import { translate } from '../../../helpers/l10n';
+import { getValues } from '../../../api/settings';
+import { FormattedMessage } from 'react-intl';
+import { formatMeasure } from '../../../helpers/measures';
+import { Link } from 'react-router';
 
 interface Props {
   branches: Branch[];
+  canAdmin?: boolean;
```

Comment:
In my PR I added `canAdmin` to the context, I think it would be nice to use it here and drop the `AppContainer`. But I understand that it's a bit complicated since it's not merged yet.

Reply:
Maybe we should do a small dedicated sprint in 6.7 or 7.0 to clean all these things up.

Reply:
Yeah maybe, do you have a list of things to clean in mind ?

Reply:
well, all these redux-vs-context things